open_event/api/events.py

```diff
@@ -38,14 +38,59 @@
 del EVENT_POST['id']
 
 
-# Helper function to fix datetime event payload
-# TODO: Make ServiceDAO more versatile so that event can use it
-def fix_payload(data):
-    data['start_time'] = EVENT_POST['start_time'].from_str(data['start_time'])
-    data['end_time'] = EVENT_POST['end_time'].from_str(data['end_time'])
-    data['closing_datetime'] = EVENT_POST['closing_datetime'].from_str(
-        data['closing_datetime'])
-    return data
+class EventDAO(ServiceDAO):
```

Comment:
It would be better to create a BaseDAO, and then extend it to EventDAO and ServiceDAO. What do you say?